playbooks/roles/nomad/tasks/main.yml

```diff
@@ -0,0 +1,100 @@
+---
+
+- name: Create nomad group
+  group:
+    name: nomad
+    system: yes
+
+- name: Create nomad user
+  user:
+    name: nomad
+    group: nomad
+    create_home: no
+    system: yes
+
+- name: Create Nomad directories
+  file:
+    path: "{{ item.path }}"
+    state: directory
+    owner: "{{ item.owner }}"
+    group: nomad
+    mode: 0750
+  with_items:
+    - { path: "{{ nomad_config_dir }}", owner: root }
+    - { path: "{{ nomad_data_dir }}", owner: nomad }
+    - { path: "{{ nomad_tls_dir }}", owner: root}
+
+- name: Copy over Nomad private TLS content
+  copy:
+    content: "{{ item.content }}"
+    dest: "{{ item.dest }}"
+  with_items:
+    - { content: "{{ nomad_admin_cert }}", dest: "{{ nomad_tls_dir }}/admin.cert" }
+    - { content: "{{ nomad_admin_key }}", dest: "{{ nomad_tls_dir }}/admin.key" }
+    - { content: "{{ nomad_ca_cert }}", dest: "{{ nomad_tls_dir }}/ca.cert" }
+    - { content: "{{ nomad_server_cert if nomad_server else nomad_client_cert }}", dest: "{{ nomad_tls_dir }}/nomad.cert" }
+    - { content: "{{ nomad_server_key  if nomad_server else nomad_client_key }}", dest: "{{ nomad_tls_dir }}/nomad.key" }
+  no_log: true
+  notify: restart nomad
+
+- name: Download and unarchive the Nomad binary
+  unarchive:
+    src: "{{ nomad_url }}"
+    dest: "{{ nomad_bin_dir }}"
+    owner: nomad
+    group: nomad
+    mode: +x
+    remote_src: yes
+    creates: "{{ nomad_bin_dir }}/nomad"
```

Comment:
So this won't run again if we update the nomad version?  Might be worth adding a comment to the version number, so the next person trying to upgrade it doesn't get confused.

Reply:
Will add a comment :+1: